src/widget/form/chatform.cpp

```diff
@@ -1085,5 +1085,47 @@ void ChatForm::hideNetcam()
 
 void ChatForm::retranslateUi()
 {
+    QString volObjectName = volButton->objectName();
+    QString micObjectName = micButton->objectName();
     loadHistoryAction->setText(tr("Load chat history..."));
+
+    switch(callButton->objectName())
+    {
+    case QStringLiteral("green"):
```

Comment:
Short answer: You can't -> use `if (); else if(); else;`.

Long answer:
Switching the `objectName` to indicate the status is **bullshit**. We need to query the actual status of the underlying object. That means:

```c++
// this is some pseudo code, just to give an idea
// tox API should send some kind of event (callback), if an audio-/video call starts/mutes/ends/…
// We react to the status change via an event handler and set the tooltip text according to the status:
void GenericChatForm::callStatusChanged(Call::Type t, Call::Status s)
{
    QString description;

    // assuming c++11 "enum class Type {Audio, Video, …}" here
    if (t == Call::Type::Audio) {
        switch (s) {
        default: break;
        case Call::Status::Active: description = tr("Hangup and end the current call."); break;
        case Call::Status::Ended: description = tr("Start a new  call."); break;
        case Call::Status::Pending: description = tr("Accept the call.!"); break;
        }
        callButton->setToolTip(description);
    }
    else if (t == Call::Type::Video) {
        switch (s) {
        default: break;
        case Call::Status::Active: description = tr("End the current video call."); break;
        case Call::Status::Ended: description = tr("Start a new video call."); break;
        case Call::Status::Pending: description = tr("Accept the video call.!"); break;
        }
        videoCallButton->setToolTip(description);
    } else {
            // undefined call type -> must not happen
            Q_ASSERT(false);
    }

    // undefined call status -> must not happen
    Q_ASSERT(!description.isEmpty());
    }